src/data/equiv/fin.lean

```diff
@@ -37,6 +37,27 @@ def fin_two_equiv : fin 2 ≃ bool :=
   end,
   assume b, match b with tt := rfl | ff := rfl end⟩
 
+/-- The 'identity' equivalence between `fin n` and `fin m` when `n = m`. -/
+
+def fin_congr {n m : ℕ} (h : n = m) : fin n ≃ fin m :=
+⟨λ k, ⟨k.1, by { subst h, exact k.2 }⟩, λ k, ⟨k.1, by { subst h, exact k.2}⟩, by tidy, by tidy⟩
```

Comment:
Is this easier to work with if defined in terms of equiv.cast?

Reply:
Seems unlikely: the `simp` lemmas won't be by `refl`, because `fin_congr` will be using the dreaded `eq.rec`. The point of this one is that the proof `h` doesn't infect the data.

Reply:
Do we already have a version of cast specific to subtypes to avoid that infection in the more general case?

Reply:
I guess we could add 
```
def subtype_equiv_of_iff {α : Type*} {P Q : α → Prop} (h : ∀ a, P a ↔ Q a) :
  { a // P a } ≃ { a // Q a } :=
⟨λ x, ⟨x.1, (h x.1).mp x.2⟩, λ x, ⟨x.1, (h x.1).mpr x.2⟩, by tidy, by tidy⟩ 
```
I'm not finding it already there.

Reply:
Oh, it's just `subtype_equiv`.

Reply:
Okay, I use that now.